test/upgrade.js

```diff
@@ -35,13 +35,34 @@ describe("Contract Upgrades", function () {
     );
 
     TokenV2 = await ethers.getContractFactory("ArrowTokenV2");
-    arrowTokenV2 = await upgrades.upgradeProxy(arrowToken, TokenV2, {
-      call: { fn: "initializeV2", args: [52] },
-    });
   });
 
   describe("Upgrade ArrowToken to ArrowTokenV2", function () {
 
+    it("Should not be upgradable by any account but the owner", async function () {
+      expect(() => upgrades.upgradeProxy(arrowToken, TokenV2.connect(signer1.address), {
+        call: { fn: "initializeV2", args: [52] },
+      })).to.throw("invalid signer");
+
+      // Upgrade should succeed after ownership is transferred.
+      await arrowToken.transferOwnership(signer1.address);
+
+      expect(() => upgrades.upgradeProxy(arrowToken, TokenV2.connect(owner.address), {
+        call: { fn: "initializeV2", args: [52] },
+      })).to.throw("invalid signer");
+
+      expect(() => upgrades.upgradeProxy(arrowToken, TokenV2.connect(signer1.address), {
+        call: { fn: "initializeV2", args: [52] },
+      })).to.not.throw;
```

Comment:
```suggestion
      expect(() => upgrades.upgradeProxy(arrowToken, TokenV2.connect(signer1.address), {
        call: { fn: "initializeV2", args: [52] },
      })).to.not.throw;
      
       // change the ownership back to owner to continue the test unit
      await arrowToken.connect(signer1).transferOwnership(owner.address);
```

Reply:
I've changed the flow of the tests. Have a look and see if it makes sense to you.